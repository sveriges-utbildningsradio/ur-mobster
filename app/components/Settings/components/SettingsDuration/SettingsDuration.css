.wrap {
  display: flex;
  flex-direction: column;
}

.time {
  color: #2cdbec;
  font-size: 26px;
  font-weight: 500;
}

.buttonsWrap {
  margin-top: 15px;
}

.button {
  background: transparent;
  border: none;
}

.icon {
  height: 36px;
  width: 36px;
}
