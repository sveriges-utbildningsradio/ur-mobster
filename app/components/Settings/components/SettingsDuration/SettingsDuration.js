// @flow

import React, { useContext } from 'react'
import styles from './SettingsDuration.css'
import Add from '../../../../assets/plus-default.png'
import AddActive from '../../../../assets/plus-active.png'
import Remove from '../../../../assets/minus-default.png'
import RemoveActive from '../../../../assets/minus-active.png'
import { SettingsStoreContext } from '../../../../store/store'
import { FIVE_MINUTES } from '../../../../constants'

type SettingsToggleProps = {
  dataE2EDecrease: string,
  dataE2EIncrease: string,
  prefixLabel?: string,
  time: number,
  updaterFn: (time: number) => void
}

const SettingsDuration = ({
  dataE2EDecrease,
  dataE2EIncrease,
  prefixLabel,
  time,
  updaterFn
}: SettingsToggleProps) => {
  const { dispatch } = useContext(SettingsStoreContext)

  const handleClickDecrease = () => {
    if (time === FIVE_MINUTES) return

    dispatch(updaterFn(time - FIVE_MINUTES))
  }

  const handleClickIncrease = () => {
    dispatch(updaterFn(time + FIVE_MINUTES))
  }

  return (
    <div className={styles.wrap}>
      <p className={styles.time}>
        {!!prefixLabel && !!prefixLabel.length && prefixLabel}
        {time / 60} min
      </p>
      <div className={styles.buttonsWrap}>
        {/* eslint-disable no-return-assign */}
        <button
          className={styles.button}
          data-e2e={dataE2EIncrease}
          onClick={() => handleClickIncrease()}
          onKeyPress={() => handleClickIncrease()}
          type="button"
        >
          <img
            alt="Increase time 5 minutes"
            className={styles.icon}
            onBlur={e => (e.currentTarget.src = Add)}
            onFocus={e => (e.currentTarget.src = AddActive)}
            onMouseOut={e => (e.currentTarget.src = Add)}
            onMouseOver={e => (e.currentTarget.src = AddActive)}
            src={Add}
          />
        </button>
        <button
          className={styles.button}
          onClick={() => handleClickDecrease()}
          onKeyPress={() => handleClickDecrease()}
          type="button"
        >
          <img
            alt="Decrease time 5 minutes"
            className={styles.icon}
            data-e2e={dataE2EDecrease}
            onBlur={e => (e.currentTarget.src = Remove)}
            onFocus={e => (e.currentTarget.src = RemoveActive)}
            onMouseOut={e => (e.currentTarget.src = Remove)}
            onMouseOver={e => (e.currentTarget.src = RemoveActive)}
            src={Remove}
          />
        </button>
      </div>
      {/* eslint-enable no-return-assign */}
    </div>
  )
}

SettingsDuration.defaultProps = {
  prefixLabel: ''
}

export default SettingsDuration
