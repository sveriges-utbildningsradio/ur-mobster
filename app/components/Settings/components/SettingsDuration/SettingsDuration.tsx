import React, { useContext } from 'react'
import styles from './SettingsDuration.css'
import Add from '../../../../assets/plus-default.png'
import AddActive from '../../../../assets/plus-active.png'
import Remove from '../../../../assets/minus-default.png'
import RemoveActive from '../../../../assets/minus-active.png'
import { SettingsStoreContext } from '../../../../store/store'

type SettingsToggleProps = {
  prefixLabel?: string
  time: number
  updaterFn: () => void
}

const FIVE_MINUTES = 60 * 5

const SettingsDuration = ({
  prefixLabel,
  time,
  updaterFn
}: SettingsToggleProps) => {
  const { dispatch } = useContext(SettingsStoreContext)

  const handleClickDecrease = () => {
    if (time === FIVE_MINUTES) return

    dispatch(updaterFn(time - FIVE_MINUTES))
  }

  const handleClickIncrease = () => {
    dispatch(updaterFn(time + FIVE_MINUTES))
  }

  return (
    <div className={styles.wrap}>
      <p className={styles.time}>
        {!!prefixLabel.length && prefixLabel}
        {time / 60} min
      </p>
      <div className={styles.buttonsWrap}>
        <button className={styles.button} type="button">
          <img
            alt="Increase time 5 minutes"
            className={styles.icon}
            onBlur={e => (e.currentTarget.src = Add)}
            onClick={() => handleClickIncrease()}
            onFocus={e => (e.currentTarget.src = AddActive)}
            onMouseOut={e => (e.currentTarget.src = Add)}
            onMouseOver={e => (e.currentTarget.src = AddActive)}
            src={Add}
          />
        </button>
        <button className={styles.button} type="button">
          <img
            alt="Decrease time 5 minutes"
            className={styles.icon}
            onBlur={e => (e.currentTarget.src = Remove)}
            onClick={() => handleClickDecrease()}
            onFocus={e => (e.currentTarget.src = RemoveActive)}
            onMouseOut={e => (e.currentTarget.src = Remove)}
            onMouseOver={e => (e.currentTarget.src = RemoveActive)}
            src={Remove}
          />
        </button>
      </div>
    </div>
  )
}

SettingsDuration.defaultProps = {
  prefixLabel: ''
}

export default SettingsDuration
