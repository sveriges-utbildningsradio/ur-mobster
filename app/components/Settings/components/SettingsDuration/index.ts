import SettingsDuration from './SettingsDuration'

export default SettingsDuration
