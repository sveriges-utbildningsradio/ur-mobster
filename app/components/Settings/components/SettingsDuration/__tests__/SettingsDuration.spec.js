import React from 'react'
import 'jest-dom/extend-expect'
import renderWithReactIntl from '../../../../../utils/testHelpers'
import SettingsDuration from '../SettingsDuration'

describe('components/Settings/SettingsDuration', () => {
  it('renders', () => {
    const { container } = renderWithReactIntl(<SettingsDuration time={300} />)

    expect(container.firstChild).toMatchSnapshot()
  })
})
