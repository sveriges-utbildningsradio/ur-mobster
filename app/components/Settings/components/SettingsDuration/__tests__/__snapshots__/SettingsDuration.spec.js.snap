// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Settings/SettingsDuration renders 1`] = `
<div
  class="wrap"
>
  <p
    class="time"
  >
    5
     min
  </p>
  <div
    class="buttonsWrap"
  >
    <button
      class="button"
      type="button"
    >
      <img
        alt="Increase time 5 minutes"
        class="icon"
        src="test-file-stub"
      />
    </button>
    <button
      class="button"
      type="button"
    >
      <img
        alt="Decrease time 5 minutes"
        class="icon"
        src="test-file-stub"
      />
    </button>
  </div>
</div>
`;
